src/gridcoin/upgrade.cpp

```diff
@@ -177,6 +183,18 @@ void Upgrade::SnapshotMain()
     std::cout << _("Snapshot Process Has Begun.") << std::endl;
     std::cout << _("Warning: Ending this process after Stage 2 will result in syncing from 0 or an incomplete/corrupted blockchain.") << std::endl << std::endl;
 
+    // Verify a mandatory release is not available before we continue to snapshot download.
+    std::string VersionResponse = "";
+
+    if (CheckForLatestUpdate(VersionResponse, false, true))
+    {
+        std::cout << _("Unable to download a snapshot, as the wallet has detected that a new mandatory version is available for install. The mandatory upgrade must be installed before the snapshot can be downloaded and applied.") << std::endl;
+        std::cout << _("Latest Version github data response:") << std::endl;
+        std::cout << VersionResponse << std::endl;
+
+        throw std::runtime_error(_("Failed to download snapshot as mandatory client is available for download."));
```

Comment:
this line is not in the qt side of things. could you elabarote where it doesn't match?

Reply:
Sorry, I meant the line above: 
```cpp
std::cout << _("Unable to download a snapshot, as the wallet has detected that a new mandatory version is available for install. The mandatory upgrade must be installed before the snapshot can be downloaded and applied.") << std::endl;
```
Since the message passed to the exception is translated as well, why not use the above in both places? That way, someone will get the full message in the log if they launch without a console (like Windows shortcuts), and it reduces work for translators.

Reply:
i agree with doing that, there is a couple other places with same messages translated twice so i'm gonna whip something up singly that will address them all.

Reply:
Did you still want to reuse `ResetBlockchainMessages(UpdateAvailable)` for the exception message to avoid a redundant translation?